src/Buffer.cpp

```diff
@@ -8,14 +8,18 @@ std::string Buffer::popFront(int num) {
     if (mBuffer.size() <= static_cast<std::size_t>(num)) {
         return std::move(mBuffer);
     }
-    auto result = mBuffer.substr(0, static_cast<std::size_t>(num - 1));
-    mBuffer.erase(0, static_cast<std::size_t>(num - 1));
+    auto result = mBuffer.substr(0, static_cast<std::size_t>(num));
+    mBuffer.erase(0, static_cast<std::size_t>(num));
     return result;
 }
 
 void Buffer::appendBack(const std::string &str, char delimiter) {
     std::lock_guard lock(mReadWriteLock);
-    mBuffer.append(str + delimiter);
+    if (mBuffer.empty()) {
+        mBuffer.append(str);
```

Comment:
Delimiter should be used for all appendings expect the first data string.

Reply:
I suppose, this is exactly what lines 18-22 are doing.

Reply:
`mBuffer` is empty only before first append?

Reply:
Not only. `mBuffer` can be empty if all data has been read in runtime.
But in case we get a data from redirected output - yes, only.

Reply:
**Not only**. But delimeter should be omited **only** for the first data string.

Reply:
I am just also considering the case, when we get a data from stdin untered by user. In that case we can read all data and get the new one. 
As for the redirected output, `mBuffer` will be empty only before first append.

Reply:
Can you provide some example?

Reply:
Data from the standard input does not differ from the data from the file. E.g. imagine this session:

```
1s < 1⏎
1s < 2⏎
1s > 1␣2
2s < 3⏎
2s > ␣3
```

where `1s`, `2s`, ... is time steps, `<` is input, `>` is output, `⏎` is enter pressed mark, `␣` is space visualization.

This is equivalent for this:

**input**
```
1
2
3
```

**output**
```
1 2 3
```

Reply:
Got it. Thank you. I will try to rework it.